Source/MQTTnet.AspnetCore/MqttConnectionContext.cs

```diff
@@ -8,28 +8,34 @@
 using MQTTnet.AspNetCore.Client.Tcp;
 using MQTTnet.Exceptions;
 using MQTTnet.Formatter;
+using MQTTnet.Internal;
 using MQTTnet.Packets;
 using System;
 using System.IO.Pipelines;
 using System.Security.Cryptography.X509Certificates;
 using System.Threading;
 using System.Threading.Tasks;
-using MQTTnet.Internal;
 
 namespace MQTTnet.AspNetCore
 {
     public sealed class MqttConnectionContext : IMqttChannelAdapter
     {
         readonly AsyncLock _writerLock = new AsyncLock();
-        
+        readonly bool? _isOverWebSocket;
         PipeReader _input;
         PipeWriter _output;
-        
+
         public MqttConnectionContext(MqttPacketFormatterAdapter packetFormatterAdapter, ConnectionContext connection)
         {
             PacketFormatterAdapter = packetFormatterAdapter ?? throw new ArgumentNullException(nameof(packetFormatterAdapter));
             Connection = connection ?? throw new ArgumentNullException(nameof(connection));
 
+            var feature = connection.Features.Get<IHttpContextFeature>();
+            if (feature != null)
+            {
+                _isOverWebSocket = feature.HttpContext.WebSockets.IsWebSocketRequest;
```

Comment:
Just out of curiosity, there any chance that this will be not a web socket request?

Reply:
Yes, `MqttConnectionHandler` has two usage scenarios, `UseMqtt` and `MapMqtt`, for `over tcp` and `over WebSocket`.